data/rulesets.ts

```diff
@@ -2965,4 +2965,124 @@ export const Rulesets: import('../sim/dex-formats').FormatDataTable = {
 			return newSpecies;
 		},
 	},
+	datapreview: {
+		effectType: 'Rule',
+		name: 'Data Preview',
+		desc: 'When a new Pokémon switches in for the first time, information about its types, stats and Abilities is displayed to both players.',
+		onSwitchIn(pokemon) {
+			let species = this.dex.species.get(pokemon.species.name);
+			const gen = this.gen;
+
+			if (pokemon.illusion) {
+				species = this.dex.species.get(pokemon.illusion.species.name);
```

Comment:
```suggestion
			const species = pokemon.illusion?.species || pokemon.species;
			const gen = this.gen;

			if (pokemon.illusion) {
```

Actually most of this is also redundant.